### DISCUSSION TITLE
Add support for custom linux distros based on linux live and sys linux

### DISCUSSION THREAD

**Discussion by megaenderdragon** [ventoy/Ventoy#1281] [Ideas]
Just like in the title. I can't launch amogOS from ventoy because it shows: no bootfile for x64 efi. perhapse the image does not support uefi. I reached out to amogOS support and they said it is normal, becuase AmogOS is on sys linux and linux live. It doesn't support copying the iso. Because ventoy only support the drag and drop isos, it closes the doors for custom linux to be on ventoy

**Comment by ventoy** [comment]
Just test the ISO file with a VM (VMware/VirtualBox/Qemu)  connect the ISO file directly to the CD/DVD of the VM and boot in UEFI mode.
If the ISO file can boot OK, then Ventoy should boot it OK, if it can not boot in VM then you should reach out the developer that "hey, the ISO file can not boot in VM".
I think it is very clear.

**Reply by megaenderdragon** [reply]
I reached out to them. They said you need to Flash the usb. I will try it on my PC. But notę. I płace the iso in ventoy folder and it did not work. The discord server member of the amogOS project says it will work on a flashed usb, because it runs on sys linux and linux live. In ventoy it shows the error I mentioned above

**Reply by ventoy** [reply]
Did you test it with VM?

**Comment by megaenderdragon** [comment]
Yes, It worked

here is the proof

https://drive.google.com/file/d/1ehYqIpEmq8RwDPbjiyn2q0d3Y2WqZ02M/view?usp=sharing

**Reply by ventoy** [reply]
I have no access.
Where to download the ISO file for test?

**Reply by megaenderdragon** [reply]
https://www.amogos.studio/ this is the link. you press download, you download x86_x64 version. Beware: there is a rickroll

**Reply by EpicLPer** [reply]
Btw. to anyone stumbling upon this these days, don't click the amogos studio link... it links to an ad-site now.

**Comment by megaenderdragon** [comment]
meanwhile in ventoy it does not boot. 

https://drive.google.com/file/d/1qj-COWV2qbD3Lxd4FdxN7Ye-auCP8DZB/view?usp=sharing

https://drive.google.com/file/d/1qcrqYQQE3hgNG1fqhYLln1iUMt-NMh-C/view?usp=sharing

**Comment by ventoy** [comment]
I download `AmogOS-1.4.0-x86_64.iso` with `SHA256: 4C180BE36E2B3448A1CF4210C0F2A5AEA118ADDF0FB6926E97ACAE7CEBD2DCA6
` 
I test with VMware and VirtualBox, Both can ***NOT*** boot in UEFI mode.

***VirtualBox***
![image](https://user-images.githubusercontent.com/59477474/143887851-c681bdbe-2a08-447c-899c-ace853132884.png)


***VMWare***
![image](https://user-images.githubusercontent.com/59477474/143887787-9f2993c3-1b2d-497f-87f7-f34c424fe025.png)

When I use the same way to test e.g. Ubuntu ISO file both can boot OK in UEFI mode.
So the ISO file itself has problem,  it can not boot in UEFI mode from CD/DVD.

**Reply by megaenderdragon** [reply]
Weird, on my it boots. How to change boot option uefi to bios so I can check?

**Reply by megaenderdragon** [reply]
How to check which way the vm boots? legacy or uefi?

**Reply by megaenderdragon** [reply]
Well, I dug around with oracle vb. same result

**Comment by NoozAbooz** [comment]
@ventoy Hey, I'm the OS developer. We currently package with https://linux-live.org/, which does support UEFI I believe. It boots fine on my UEFI laptop and PC.

**Reply by ventoy** [reply]
There should be an `EFI` directory in the root of the ISO file.
This is a common flag for those ISO files who support UEFI boot. (Ubuntu/Arch/Redhat/Suse ......  you can check them)
But there is only a `linux` directory in  your ISO.

**Reply by NoozAbooz** [reply]
> There should be an `EFI` directory in the root of the ISO file. This is a common flag for those ISO files who support UEFI boot. (Ubuntu/Arch/Redhat/Suse ...... you can check them) But there is only a `linux` directory in your ISO.

I believe there's a EFI directory inside the linux directory.

**Reply by ventoy** [reply]
Not inside the linux directory, it should be in the root of the ISO file. 
When you boot in UEFI mode, the UEFI  BIOS fixed launch the  `\EFI\BOOT\BOOTX64.EFI` file.

**Reply by ventoy** [reply]
`\EFI\BOOT\BOOTX64.EFI` the path is fixed and defined in UEFI specification.

**Reply by NoozAbooz** [reply]
> 

Well then, that might be something to ask the linux live developers

**Reply by megaenderdragon** [reply]
> @ventoy Hey, I'm the OS developer. We currently package with https://linux-live.org/, which does support UEFI I believe. It boots fine on my UEFI laptop and PC.

Maybe because you use the legacy boot Mode, even though your PC can support uefi

**Comment by NoozAbooz** [comment]
@ventoy I assume you probably know alot about packaging tools and stuff since ventoy is a multiboot app. Do you know any good packaging tools for packaging a linux distro to a ISO other than linux live?

**Reply by ghost** [reply]
Would PinguyBuilder do? I know it works for vanilla Ubuntu 20.04
(there is a size limit, to 4GB)

**Reply by NoozAbooz** [reply]
> Would PinguyBuilder do? I know it works for vanilla Ubuntu 20.04 (there is a size limit, to 4GB)

Size shouldn't be a problem, github limits file sizes to 2gb anyways. Not sure if it would work tho considering AmogOS is debian-based.
